server/app/mutations/stacks/create.rb

```diff
@@ -45,11 +51,28 @@ def validate_services
       add_error("services.#{error.service}.links", :recursive, error.message)
     end
 
+    def validate_parent
+      _, parent_name = if self.inputs.has_key?(:parent)
+        self.inputs.delete(:parent)[:id].split('/')
+      else
+        [self.grid.name, self.inputs.delete(:parent_name)]
+      end
+      return unless parent_name
+
+      parent_stack = self.grid.stacks.find_by(name: parent_name)
+      if parent_stack.nil?
+        add_error(:parent, :not_found, "parent stack #{parent_name} does not exist")
+      elsif parent_stack.grid_id != self.grid.id
+        add_error(:parent, :not_found, "parent stack #{parent_name} does not exist")
+      else
+        @parent = parent_name
+      end
+    end
+
     def execute
       attributes = self.inputs.clone
       grid = attributes.delete(:grid)
-      parent = attributes.delete(:parent_name)
```

Comment:
```

Mongoid::Errors::UnknownAttribute (
message:
  Attempted to set a value for 'parent_name' which is not allowed on the model StackRevision.
summary:
  Without including Mongoid::Attributes::Dynamic in your model and the attribute does not already exist in the attributes hash, attempting to call StackRevision#parent_name= for it is not allowed. This is also triggered by passing the attribute to any method that accepts an attributes hash, and is raised instead of getting a NoMethodError.
resolution:
  You can include Mongoid::Attributes::Dynamic if you expect to be writing values for undefined fields often.):
  /usr/lib/ruby/gems/2.4.0/gems/mongoid-5.2.1/lib/mongoid/attributes/processing.rb:99:in `process_attribute'
  /usr/lib/ruby/gems/2.4.0/gems/mongoid-5.2.1/lib/mongoid/attributes/processing.rb:25:in `block in process_attributes'
  /usr/lib/ruby/gems/2.4.0/gems/mongoid-5.2.1/lib/mongoid/attributes/processing.rb:23:in `each_pair'
  /usr/lib/ruby/gems/2.4.0/gems/mongoid-5.2.1/lib/mongoid/attributes/processing.rb:23:in `process_attributes'
  /usr/lib/ruby/gems/2.4.0/gems/mongoid-5.2.1/lib/mongoid/document.rb:111:in `block in initialize'
  /usr/lib/ruby/gems/2.4.0/gems/mongoid-5.2.1/lib/mongoid/threaded/lifecycle.rb:89:in `_building'
  /usr/lib/ruby/gems/2.4.0/gems/mongoid-5.2.1/lib/mongoid/document.rb:105:in `initialize'
  /usr/lib/ruby/gems/2.4.0/gems/mongoid-5.2.1/lib/mongoid/factory.rb:25:in `new'
  /usr/lib/ruby/gems/2.4.0/gems/mongoid-5.2.1/lib/mongoid/factory.rb:25:in `build'
  /usr/lib/ruby/gems/2.4.0/gems/mongoid-5.2.1/lib/mongoid/relations/referenced/many.rb:82:in `build'
  /usr/lib/ruby/gems/2.4.0/gems/mongoid-5.2.1/lib/mongoid/relations/many.rb:77:in `create!'
  /app/app/mutations/stacks/create.rb:87:in `execute'
```

Reply:
I see, this didn't show up in the specs because it only happened after I commented out the `#validate_parent`, which does `self.inputs.delete(:parent_name)` as a side-effect, preventing this.